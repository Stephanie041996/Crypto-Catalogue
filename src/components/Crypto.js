import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { GetCrypto } from '../actions/cryptoActions';
import _ from "lodash";

function Crypto(props){
    const cryptoId = props.match.params.id;
  const dispatch = useDispatch()
  const cryptoState = useSelector(state => state.id)
  React.useEffect( () => {
    dispatch(GetCrypto(cryptoId))
  },[cryptoId, dispatch]);
console.log( cryptoId)

    


  return(
    <div className={"poke"}>
      <h1>{cryptoId}</h1>
     
    </div>
  )
};


export default Crypto