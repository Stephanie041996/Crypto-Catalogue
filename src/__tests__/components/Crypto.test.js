import React from 'react';
import renderer from 'react-test-renderer';
import { render, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Crypto from '../../components/Crypto';
import store from '../../Store';

afterEach(cleanup);

it('should render the list of cryptos', () => {
  const match = { params: { id: 'bitcoin' } };
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Crypto match={match} />
      </BrowserRouter>
    </Provider>,
  );
  expect(
    asFragment(
      <Provider store={store}>
        <Crypto match={match} />
      </Provider>,
    ),
  ).toMatchSnapshot();
});

describe('Crypto DOM', () => {
  const match = { params: { id: 'bitcoin' } };
  it('renders crypto content correctly', () => {
    const Cryp = renderer.create(
      <Provider store={store}>
        <Crypto match={match} />
        ,
      </Provider>,
    ).toJSON();
    expect(Cryp.name).not.toBe('random');
  });
});
