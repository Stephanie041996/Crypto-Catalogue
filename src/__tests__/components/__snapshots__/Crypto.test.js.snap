// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the list of cryptos 1`] = `
<DocumentFragment>
  <div
    class="cryp"
  >
    <h2>
      Crypto Info
    </h2>
    <h2>
      <p>
        ERRROR
      </p>
    </h2>
  </div>
</DocumentFragment>
`;
